Fix course theme comment styles

Imran92 opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:

Fixes discrepancies in the comment thread and reply form styles among different templates, like posts, page, lesson, and optional templates.

### Testing instructions

1. Enable the Course theme
2. Enable Sensei
3. Create a page and a post, make sure to enable comments
4. Create a Course with a lesson in sensei, and enable comments for it too
5. Now visit the page and post from the frontend as a logged out user
6. Make a few comments
7. Make sure the thread and the reply forms look close to the design
8. Now open the post in the editor
9. Select each custom template from the right dropdown, like Single Column - Featured Image Only or Single Column - Title only, etc.
10. After selecting each of them, click on update and check from the frontend. Make sure the comments are not broken in any of them.
11. Now check them again as logged in user.
12. You won't see all the fields now, but make sure it doesn't appear broken.
13. Make sure to check for the lesson now too. Add a couple of comments, make sure the thread looks like the other ones from page and post type.

#### For now, disregard if you see the comment portion getting duplicated in lessons. It'll be fixed in a separate PR under a different issue. Only check for the comment portion at the bottom, not the top one.

We tried reusing the existing spacing and size presets to not introduce new ones. So in some places, the spacings are few pixels different, but they don't fall far from the design. Also, the design has some spacings that are not uniform, like the spacing between all the input fields or between the input fields and their labels, like the spacings between the email label and its input field and the comment label and its input field. We used a uniform spacing in these places.

![image](https://github.com/Automattic/themes/assets/6820724/d26c0d1e-249f-4eee-97cd-3efd3ec0d276)


### Screenshots

Desktop:
![image](https://github.com/Automattic/themes/assets/6820724/5a911f98-5641-4e98-9365-c33a8fd080de)

Mobile:
![image](https://github.com/Automattic/themes/assets/6820724/6192dd6c-1da5-4ac8-a1d8-351088c9077d)

Design:
![image](https://github.com/Automattic/themes/assets/6820724/970178be-3909-41d9-81f7-8625fdc09fc5)

#### Related issue(s):

https://github.com/Automattic/themes/issues/6977

Imran92 committed:
Update comment pattern to match design

Imran92 committed:
Update comment styles and add pseudo checkbox style

Imran92 committed:
Update single page and lesson template to use comment pattern

Imran92 committed:
Update theme.json to match comment reply design

Imran92 committed:
Update single column templates to use comment pattern

Imran92 committed:
Update templates to use common comment pattern

donnapep commented:
When Learning Mode is disabled, the lessons page looks pretty terrible. Let me know if you think we should create a separate card for this:

![Screenshot 2023-07-24 at 8 26 01 AM](https://github.com/Automattic/themes/assets/1190420/3dd2c54e-9f82-4fd1-915c-34156e1cb3f7)

Imran92 commented:
> For now, disregard if you see the comment portion getting duplicated in lessons. It'll be fixed in a separate PR under a different issue. Only check for the comment portion at the bottom, not the top one.

Hi @donnapep 👋 Thanks for taking a look. The one you are seeing is the screenshot is the comment section that's being automatically generated by sensei and not by the theme, so it's actually the 'duplicate' one that we see in the FSE themes. It's being worked on under this issue https://github.com/Automattic/sensei/issues/7037 already. It's also mentioned in the PR description above

Thanks!

donnapep commented on course/patterns/comments.php:
```diff
@@ -8,42 +8,41 @@
 <!-- wp:comments {"className":"wp-block-comments-query-loop "} -->
-<div class="wp-block-comments wp-block-comments-query-loop">
-<!-- wp:comments-title {"showPostTitle":false,"showCommentsCount":false,"level":3,"style":{"typography":{"fontStyle":"normal","fontWeight":"600"}},"fontSize":"large"} /-->
+<div class="wp-block-comments wp-block-comments-query-loop"><!-- wp:comments-title {"showPostTitle":false,"showCommentsCount":false,"level":3,"style":{"typography":{"fontStyle":"normal","fontWeight":"600","lineHeight":"1"},"spacing":{"padding":{"top":"0","right":"0","bottom":"40px","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}},"fontSize":"large"} /-->
```
Is this `40px` adding whitespace between 2 elements? If so, that should be margin instead of padding.

donnapep reviewed (commented):
Here's some things I found while testing:

1. The size of the avatar for post comments (50 x 50) is different than it is for page and lesson comments (42 x 42). The post comments is closer to the design, which is 48 x 48 including the border. (But 50 x 50 is also 🆗 so long as they're all consistent.)
2. The space between the avatar and the username / date is also inconsistent (should be 20px for all).
3. There's extra whitespace between the line and the _Responses_ heading for posts. There's also a bit extra below (`40px` vs `36px` for pages / lessons):

![Screenshot 2023-07-24 at 8 41 21 AM](https://github.com/Automattic/themes/assets/1190420/0c034e51-69d8-46ce-bdd9-2aa605f3825a)

As compared to pages or lessons, which has `36px` top and bottom:

![Screenshot 2023-07-24 at 8 42 01 AM](https://github.com/Automattic/themes/assets/1190420/a2663023-690b-4d2f-bed6-514ce9a80b06)
In the design the top and bottom margins are `48px`, but again, I'm OK with `36px` as long as we're consistent across all post types.
4. Paged comments don't look too good. Although there's no design for it, could we move pagination closer to the comments and put the `80px` of bottom margin between it and "Leave a Reply"?

![Screenshot 2023-07-24 at 9 01 49 AM](https://github.com/Automattic/themes/assets/1190420/f78ab710-0344-40d0-bf81-0d12e51d2f9a)

I haven't gone over all the details of the design with a fine-toothed comb, so please consider this review incomplete. Mostly I'm looking for the comment styles to be consistent between posts, pages and lessons and to generally look good. I'll go over it more closely once you've given it another pass. 🙂

Imran92 committed:
Move top border to comment pattern

Imran92 committed:
Add preset for top and bottom margin of comment title

Imran92 committed:
Add spacings for pagination and remove spacer

Imran92 commented:
Hi @donnapep 👋 Thanks so much for checking!

1. 

> The size of the avatar for post comments (50 x 50) is different than it is for page and lesson comments (42 x 42). The post comments is closer to the design, which is 48 x 48 including the border. (But 50 x 50 is also 🆗 so long as they're all consistent.)

I couldn't reproduce this. Can you kindly check if your local environment is using this branch and there is no saved customization for any template from previous times? We are using one common pattern file and are using its reference from all other templates. So it should be consistent all over, in case there isn't targeted CSS for specific pattern or post type. I tried with different browsers, incognito mode etc. Here are some screenshots from my env
Page:
![Page 48](https://github.com/Automattic/themes/assets/6820724/3696e448-6e73-46cb-9418-b357f4d35263)
Post:
![Post 48](https://github.com/Automattic/themes/assets/6820724/63e6dc45-2d85-4b98-adea-db9763c1e09e)
Lesson:
![Lesson 48](https://github.com/Automattic/themes/assets/6820724/446f5ee1-8846-4c94-8a5a-e89fcf14d230)

2. 

> The space between the avatar and the username / date is also inconsistent (should be 20px for all).

Same as above, I couldn't reproduce this issue. It should be same because of using same pattern file. Here are my screenshots (I've used 20px column gap, the scale extension picks 22px). The dates also look the same in the screenshots
Page:
![Page 22](https://github.com/Automattic/themes/assets/6820724/6aa5ddbe-e7eb-4074-bf46-9e8b446d7ffd)
Post:
![Post 22](https://github.com/Automattic/themes/assets/6820724/a729c04c-802d-4667-86f2-a62fa902095f)
Lesson:
![Lesson 22](https://github.com/Automattic/themes/assets/6820724/99ff4a1c-5392-4866-90f0-8d3d6256008f)

3.

> There's extra whitespace between the line and the Responses heading for posts. There's also a bit extra below (40px vs 36px for pages / lessons):

Thanks! I've changed it here https://github.com/Automattic/themes/pull/7252/commits/7ade40da55c6454780abfc9c6979e3147b29bb76.

4. 

> Paged comments don't look too good. Although there's no design for it, could we move pagination closer to the comments and put the 80px of bottom margin between it and "Leave a Reply"?

Good suggestion. The pagination looked a bit out of place to me too. I've updated the spacings here https://github.com/Automattic/themes/pull/7252/commits/dafffcc548bcc2f333f04c4b503863f59c2ad67a

5. 

> Is this 40px adding whitespace between 2 elements? If so, that should be margin instead of padding.

Thanks for the good suggestion, updated here https://github.com/Automattic/themes/pull/7252/commits/dafffcc548bcc2f333f04c4b503863f59c2ad67a

Thanks again for taking a look @donnapep ! Please give it another check when you can :) !

donnapep commented on course/theme.json:
```diff
@@ -137,6 +137,7 @@
 			"commentGap" : "5rem",
 			"commentGapSmall" : "10px",
 			"commentGapMedium" : "1.25rem",
+			"commentTopMargin": "clamp(2.5rem, 2.167rem + 1.111vw, 3rem)",
```
I think custom presets are meant to be used when we're using their values in multiple places. It doesn't appear that we're doing that here so I'd put it right in the CSS.

donnapep reviewed (commented):
@Imran92 Thanks! 1 & 2 are resolved. It does look like I had a customization. 😳 

For number 3, I still see some extra whitespace. In the screenshot, the `.wp-block-group` element has 40px of margin, then there's a spacer that is `4rem`, and then `.wp-block-comments` also has 40px of margin.

![Screenshot 2023-07-27 at 12 31 07 PM](https://github.com/Automattic/themes/assets/1190420/a611280f-bb9a-4c5d-96bf-7e265e947580)

The pagination isn't fixed for me either:

![Screenshot 2023-07-27 at 12 33 57 PM](https://github.com/Automattic/themes/assets/1190420/f866d660-5990-4702-af76-65687a5c9e48)

Everything else is looking great!

donnapep commented on course/patterns/comments.php:
```diff
@@ -8,42 +8,41 @@
-<!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","orientation":"vertical"}} -->
-<div class="wp-block-group" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:comment-author-name /-->
+	<!-- wp:comment-template {"style":{"spacing":{"padding":{"top":"0","right":"0","bottom":"0","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}}} -->
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"0","right":"0","left":"0","bottom":"5rem"},"blockGap":"0"}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap"}} -->
```
Should this be margin instead of padding too?

Imran92 committed:
Remove extra class from pagination and move preset style to css

Imran92 committed:
Use margin instead of padding

Imran92 commented:
> @Imran92 Thanks! 1 & 2 are resolved. It does look like I had a customization. 😳

Whew! Glad it worked! ☺️ Thanks for finding out the customization.

> For number 3, I still see some extra whitespace. In the screenshot, the .wp-block-group element has 40px of margin, then there's a spacer that is 4rem, and then .wp-block-comments also has 40px of margin.

I couldn't reproduce this one. For me, all of page, post and lesson had 48px (the scale extension picks 49) above and below the Responses title. Can there be any issue like customization or caching here? Here are some screenshots from my environment. And just to make sure, I've tested with every template in case you had a different one. But still didn't find any issue

Page:
![Screenshot 2023-07-27 at 10 51 45 PM](https://github.com/Automattic/themes/assets/6820724/ef2848ef-eb61-4bd2-b352-e96c2ee6bf06)

Post:
![Screenshot 2023-07-27 at 10 52 18 PM](https://github.com/Automattic/themes/assets/6820724/477c0d2d-410d-46d3-8f99-b5c6f168e173)

Lesson:
![Screenshot 2023-07-27 at 10 53 07 PM](https://github.com/Automattic/themes/assets/6820724/affab7fc-1e65-4233-9552-9f0535ed97e9)

> The pagination isn't fixed for me either:

Fixed it here https://github.com/Automattic/themes/pull/7252/commits/a39ede88778a7dbb8f77b863eb679a21fa855136 ☺️  Should work now!

> I think custom presets are meant to be used when we're using their values in multiple places. It doesn't appear that we're doing that here so I'd put it right in the CSS.

Yap, you're right. Used the spacing directly in CSS here https://github.com/Automattic/themes/pull/7252/commits/a39ede88778a7dbb8f77b863eb679a21fa855136

> Should this be margin instead of padding too?

Changed it to margin here https://github.com/Automattic/themes/pull/7252/commits/ea225bedad4058b620f158d45bdfbc9c2d4cc95d

Thanks @donnapep ! Please take a look again when possible

donnapep commented:
I figured out what the problem was (sync issue) and am no longer seeing the previous issues. 🙌🏻 

After doing a bit more extensive testing, I found one other issue when replying to threaded comments:

![Screenshot 2023-07-31 at 8 48 39 AM](https://github.com/Automattic/themes/assets/1190420/b903c28b-a599-4d43-aaa7-31b5dc5b74b9)

- We should style the _Cancel Reply_ link so that it looks the same as the _Edit_ link.
- Is it possible to make the text in the textbox `var(--wp--preset--color--foreground)` instead of black?
- There needs to be a bit more whitespace between the _Post Comment_ button and the following comment. `5rem` would match the current spacing between comments.

donnapep commented:
Also, given the Themes team will end up deploying our changes along with any changes they have ready to ship as soon as we merge to `trunk`, I think we should consider putting these changes into a feature branch and, depending on how much we accomplish, maybe push a new release halfway through the project. Perhaps releasing just the theme, or also Sensei if there are any dependencies (which we should be sure to track). Thoughts?

donnapep commented on course/style.css:
```diff
@@ -127,6 +127,61 @@ h1, h2, h3, h4, h5, h6 {
+
+.wp-block-comments-query-loop .wp-block-comments-pagination {
+	margin-top: calc( var(--wp--custom--comment-gap) / 2 * -1 );
+	padding-bottom: var(--wp--custom--comment-gap);
```
```suggestion
	margin-bottom: var(--wp--custom--comment-gap);
```

donnapep commented on course/patterns/comments.php:
```diff
@@ -8,42 +8,41 @@
-<div class="wp-block-group" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:comment-author-name /-->
+	<!-- wp:comment-template {"style":{"spacing":{"padding":{"top":"0","right":"0","bottom":"0","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}}} -->
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"0","right":"0","left":"0","bottom":"0rem"},"blockGap":"0","margin":{"bottom":"5rem"}}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap"}} -->
+	<div class="wp-block-group" style="margin-bottom:5rem;padding-top:0;padding-right:0;padding-bottom:0rem;padding-left:0"><!-- wp:group {"style":{"spacing":{"blockGap":"20px","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left","verticalAlignment":"top"}} -->
```
```suggestion
	<div class="wp-block-group" style="margin-bottom:5rem;padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:group {"style":{"spacing":{"blockGap":"20px","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left","verticalAlignment":"top"}} -->
```

donnapep commented on course/patterns/comments.php:
```diff
@@ -8,42 +8,41 @@
-<!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","orientation":"vertical"}} -->
-<div class="wp-block-group" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:comment-author-name /-->
+	<!-- wp:comment-template {"style":{"spacing":{"padding":{"top":"0","right":"0","bottom":"0","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}}} -->
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"0","right":"0","left":"0","bottom":"0rem"},"blockGap":"0","margin":{"bottom":"5rem"}}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap"}} -->
```
Can just be `0` instead of `0rem` for padding.

donnapep commented on course/style.css:
```diff
@@ -127,6 +127,61 @@ h1, h2, h3, h4, h5, h6 {
+	margin-bottom: clamp(2.5rem, 2.167rem + 1.111vw, 3rem);
+}
+
+.wp-block-post-comments-form .comment-form-cookies-consent input[type="checkbox"] {
```
How can I test the cookies consent form?

Imran92 replied:
It is this checkbox on the bottom that you see when logged out 
<img width="854" alt="image" src="https://github.com/Automattic/themes/assets/6820724/a940a3f6-e3f9-4dfa-90db-f3832dfa87e5">

Imran92 committed:
Update course/style.css

Co-authored-by: Donna Peplinskie <donnapep@gmail.com>

Imran92 commented:
> I think we should consider putting these changes into a feature branch and, depending on how much we accomplish, maybe push a new release halfway through the project.

Makes sense to me, we can keep track of which version of the theme received the updates pertaining to the frontend improvement projects this way

Imran92 committed:
Change 0 rem to 0

Imran92 committed:
Add spacing after comment reply form in middle

Imran92 committed:
Resize cancel reply link text

Imran92 committed:
Change color of input text in comment reply form fields

Imran92 commented:
Hi @donnapep , Thanks for the comments!

> We should style the Cancel Reply link so that it looks the same as the Edit link.

Updated here https://github.com/Automattic/themes/pull/7252/commits/1eb1439a18027a33972c4874620deb4e62fd62c5

> Is it possible to make the text in the textbox var(--wp--preset--color--foreground) instead of black?

Updated the color here https://github.com/Automattic/themes/pull/7252/commits/82cca3ac0c5c4c5af4bfa04e935dd8a571c8766d

> There needs to be a bit more whitespace between the Post Comment button and the following comment. 5rem would match the current spacing between comments.

Added space here https://github.com/Automattic/themes/pull/7252/commits/57d40fc7f4d5dfb1b24d24c24c9b5cd9e6c5af59

> Can just be 0 instead of 0rem for padding.

Changed it here https://github.com/Automattic/themes/pull/7252/commits/deca37e0e1ef49ba026b1d5d3ba2ca84f3e67a08

Also, I want to ask your opinion about a style. When writing inside the comment reply form fields, the text size caused some eye strain for me because of being too small. Did you feel the same way? Maybe we can increase the font size a bit if you felt the same?

I've created a feature/frontend-improvements branch as per our discussion. I'll change the target branch from base to our feature branch for this PR.

Thanks again for checking. Please take another look when you can!

donnapep commented:
> Also, I want to ask your opinion about a style. When writing inside the comment reply form fields, the text size caused some eye strain for me because of being too small. Did you feel the same way? Maybe we can increase the font size a bit if you felt the same?

Yup, I agree. The `x-small` size looked pretty good when I tried it. Maybe we could do it for all these reply form fields actually:

![Screenshot 2023-08-02 at 10 43 16 AM](https://github.com/Automattic/themes/assets/1190420/d7864e14-7bc1-4954-800a-dd1bc1f826d8)

> I've created a feature/frontend-improvements branch as per our discussion. I'll change the target branch from base to our feature branch for this PR.

As per our 1:1 chat, we'll plan to release this one next week, so we can keep the branch at `trunk` and merge when we're ready to release.

Imran92 committed:
Update font size of comment input texts

Imran92 commented:
Hi @donnapep 👋 I've updated the comment text size here https://github.com/Automattic/themes/pull/7252/commits/635f6a7ac46e5e7367a0ad6168e83e77e592fbf7 to x-small, looks good to me too. Have also changed the target branch to trunk.

donnapep reviewed (approved):
Looks great! 🥳
